### DISCUSSION TITLE
[Question] Performance & RAM usage

### DISCUSSION THREAD

**Discussion by ReimuNotMoe** [markqvist/Reticulum#438] [Help & Questions]
Hi, I would like to get Reticulum running on a MT7628 router board with 64MB RAM. Do you think the CPU performance & RAM size are enough?

**Comment by faragher** [comment]
No. The current Python implementation, at idle, is around 400 megs according to top. I had a 1 gig server that ran well until a file transfer was requested, then it would crash due to lack of memory. Upgrading to 4 gigs shows it peaks just under 2 gigs. So for a simple router, I'd say 1 gig RAM is about the minimum, maybe 512 with a super lean OS. 2-4 gigs is the best bet for something running a node.

Other implementations aren't complete, and we can't judge their memory footprint at this time. I expect them to be lower once mature, but until they're complete, it's not very helpful to speculate.

**Comment by markqvist** [comment]
I think the 400 megabytes at idle would be the *virtual memory*; all memory areas mapped into the virtual address space of the process.

The *resident memory*, the actual amount of memory used specifically by the process, sits at around 37 megabytes for an idle `rnsd` on my system (with only a couple of interfaces). This number will obviously go up as you add interfaces and path tables grow, so 64 megs of RAM will still be to small, currently.

Memory usage can be optimised *a lot* though, and I hope to be able to do that in the relatively near future.

**Reply by faragher** [reply]
This is a valid correction. Much of the reported memory consumption is not truly committed, and I've had RNS and NomadNet basically sit in the same ~500 MB block. However, it does terminate for one reason or another on that one gig server (which is likely something that can be corrected anyway, I am simply reporting my experiences)
